umad/src/main/java/com/pkukielka/AccessMonitorRewriter.java

```diff
@@ -120,8 +117,12 @@ protected void editMethod(CtMethod editableMethod, String ifCalledFrom, String d
     }
 
     @Override
-    public void applyOnMethod(CtMethod editableMethod, String dottedName) throws CannotCompileException {
-        if (!isSafe(editableMethod, dottedName)) super.applyOnMethod(editableMethod, dottedName);
+    public void applyOnMethod(CtMethod editableMethod, String dottedClassName) throws CannotCompileException {
+        int safetyChecksRecursionLevel = 1;
+        if (config.hasPath("safetyChecksRecursionLevel"))
```

Comment:
Can we read it only once (in constructor) not for each method? Umad is slow enough :)